### PR TITLE
enh: Create a growable Resnet module

### PR DESCRIPTION
**Pull request by TheoRudkiewicz**
Introduce a `SequentialGrowingContainer` structure specialized for container with sequential layers.
Introduce a `ResNetBasicBlock` class to create resnet 18/34 like structure with growable blocks and the possibility of adding blocks.

### CODE REVIEW

**Review by sylvchev — APPROVED** [review]
Nice PR!

File: src/gromo/containers/resnet.py [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+                nn.Conv2d(
+                    in_features,
+                    inplanes,
+                    kernel_size=3,
+                    stride=1,
+                    padding=1,
+                    bias=False,
+                    device=self.device,
+                ),
+                # nn.BatchNorm2d(inplanes, device=self.device),
```
[the remark this answers is not in the document]
Comment (TheoRudkiewicz) [reply src/gromo/containers/resnet.py]: I think no. In the standard architecture there is one but here I use the full pre-activation architecture of the first operation of the block is Batchnorm and activation. So having an BN and activation before the blocks would be redundant (which in the case of ReLU is not a problem but is one otherwise).

File: src/gromo/containers/resnet.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+        return (
+            layer.out_features - int(layer.out_features * self.reduction_factor)
+        ) // growth_step
+
+
+def init_full_resnet_structure(
+    input_shape: tuple[int, int, int] = (3, 224, 224),
+    in_features: int | None = None,
+    out_features: int = 1000,
+    device: torch.device | str | None = None,
```
[the remark this answers is not in the document]
Comment (TheoRudkiewicz) [reply src/gromo/containers/resnet.py]: I just took the same as the default ResNet architecture that is designed for ImageNet (so 1000 classes).
